lorawan-device/src/async_device/mod.rs

```diff
@@ -560,39 +560,38 @@ where
         rng: &mut G,
         response: mac::Response,
     ) -> Result<Option<mac::Response>, Error<R::PhyError>> {
+        radio_buffer.clear();
         match response {
-            mac::Response::NoUpdate => {
-                radio_buffer.clear();
-                Ok(None)
-            }
+            mac::Response::NoUpdate => Ok(None),
```

Comment:
That looks odd.. :)

Reply:
How do you mean? We could probably have the mac layer return `Option<mac::Response>` and we could get rid of  `mac::Response::NoUpdate`... but I'm calling that out of scope for this PR :laughing: 

Or are you talking about how `radio_buffer.clear()` has been moved above the match?

Reply:
No, that line is duplicated.

Reply:
ah good catch! thanks